File: API/TaskMaster/TaskMaster/Model/Domain/ProjectData/BugReport.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

using TaskMaster.Model.Domain.UserData;

namespace TaskMaster.Model.Domain.ProjectData;

public class BugReport
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int BugReportId { get; set; }

    [Required]
    public string Title { get; set; }

    [Required]
    public string Details { get; set; }

    public string Notes { get; set; }
    [Required]
    public int Priority { get; set; }

    [Required]
    public int Severity { get; set; }

    [Required]
    public bool Fixed { get; set; }
    
    public DateTime? DateFixed { get; set; }

    [Required]
    [ForeignKey("User")]
    public int UserId { get; set; }

    [Required]
    [ForeignKey("Project")]
    public int ProjectId { get; set; }

    public User User { get; set; }
    public Project Project { get; set; }
}

File: App/TaskMaster/TaskMaster/Models/BugReport.swift
//
//  BugReport.swift
//  TaskMaster
//
//  Created by Joshua Root on 12/4/23.
//

import SwiftData
import Foundation

@Model
class BugReport {
    var title: String
    var details: String
    var notes: String
    var priority: Int
    var severity: Int
    var fixed: Bool
    var dateFixed: Date?
    var project: Project

    

    init(title: String, details: String, notes: String, priority: Int, severity: Int, fixed: Bool, dateFixed: Date?, user: User, project: Project) {
        self.title = title
        self.details = details
        self.notes = notes
        self.priority = priority
        self.severity = severity
        self.fixed = fixed
        self.dateFixed = dateFixed
        self.user = user
        self.project = project
    }

    
}
